File: backend/src/routes/me.ts
import { createClient } from "@supabase/supabase-js";
import { Router } from "express";
import { settings } from "../config.js";
import { requireAuth, type AuthedRequest } from "../middleware/auth.js";

export const meRouter = Router();

/** Proves the auth wiring end to end: given a caller's JWT, ask Supabase who they are. */
meRouter.get("/me", requireAuth, async (req, res) => {
  const { token } = req as AuthedRequest;
  const client = createClient(settings.supabaseUrl, settings.supabaseAnonKey);

  const { data, error } = await client.auth.getUser(token);
  if (error || !data.user) {
    res.status(401).json({ detail: "Invalid or expired session" });
    return;
  }

  res.json({ id: data.user.id, email: data.user.email ?? null });
});

File: backend/app/routers/me.py
from fastapi import APIRouter, Depends, HTTPException, status
from supabase import create_client

from ..config import settings
from ..deps import get_bearer_token

router = APIRouter()


@router.get("/me")
def read_current_user(token: str = Depends(get_bearer_token)) -> dict[str, str | None]:
    """Proves the auth wiring end to end: given a caller's JWT, ask Supabase who they are."""
    client = create_client(settings.supabase_url, settings.supabase_anon_key)
    try:
        response = client.auth.get_user(token)
    except Exception as exc:
        raise HTTPException(status.HTTP_401_UNAUTHORIZED, "Invalid or expired session") from exc

    user = response.user
    return {"id": user.id, "email": user.email}
